### ISSUE TITLE
[bug] deep-linking causes "could not add user. Please try again later"

### ISSUE THREAD

**Issue by galeone** [python-telegram-bot/python-telegram-bot#1255]
I'm trying to add a parameter to the /start command of my bot, in order of being able to link it to a user and make the user starts monitoring a product (it's an Amazon price monitor bot).
The address is something like: `t.me/<name of the bot>?start=<base64encodedparameter>`

I've implemented server side the proper handler (that's just the same handler I use for the /monitor command that works without any problem). However, when I try to use deep linking to start a conversation with the bot:

- Telegram Desktop (Linux) raises the error: "could not add user. Please try again later"
- Telegram for Android does nothing

The problem is something related to python-telegram-bot deep-linking handling, probably. I say so because the same behavior ("could not add user. Please try again later" / no response) arises even if I try to start a conversation with the bot you linked in the wiki: https://github.com/python-telegram-bot/python-telegram-bot/wiki/Types-of-Handlers#deep-linking-start-parameters

https://t.me/roolsbot?start=Hello%20World!

Same exact behavior.

Also, I've tested if the /start command I implemented works correctly without using deep-linking, and it does (the test is just to invoke `/start <base64encodedparameter>` and it monitors the product without any problem.

In short:
### Steps to reproduce

1. https://t.me/roolsbot?start=Hello%20World!

2. Press start

### Expected behaviour
It should add the bot and start the conversation, passing the parameters to the start command.

### Actual behaviour
- Telegram Desktop (Linux) raises the error: "could not add user. Please try again later"
- Telegram for Android does nothing

**Comment by jsmnbom** [comment]
This isn't a bug with our library.
Telegram simply no longer supports spaces (%20) in deep linking parameters.
It would be okay if you had actually base64 encoded the parameter though: `https://t.me/roolsbot?start=SGVsbG8gV29ybGQ=`
I'm closing this issue - feel free to track down the repositories for the official desktop and android client, and opening an issue, detailing the highly confusing error messages there :)

**Comment by galeone** [comment]
There should be something strange somewhere:

If I visit: https://telegram.me/discountposter_bot?start=aHR0cDovL3d3dy5hbWF6b24uaXQvQXBwbGUtaVBob25lLVNpbmdsZS02NEdCLUdyZXkvZHAvQjA3NVJCMTQ1WS8=

Where the start parameter value is generated with:

    base64.urlsafe_b64encode(b"http://www.amazon.it/Apple-iPhone-Single-64GB-Grey/dp/B075RB145Y/").decode('utf-8')

I get the error. But this seems the correct way of encoding a string in base64 encoder to be passed as a URL paramenter.

The same error arises when I try to pass the same string to your bot:

https://t.me/roolsbot?start=aHR0cDovL3d3dy5hbWF6b24uaXQvQXBwbGUtaVBob25lLVNpbmdsZS02NEdCLUdyZXkvZHAvQjA3NVJCMTQ1WS8=

If this is the correct way of encoding a string, where do you think the problem can be? Maybe in the lenght of the parameter?
